File: src/utils/to_camel_case.rs
pub fn to_camel_case(name: &str) -> String {
    let mut result = String::new();
    let mut first = false;
    for c in name.chars() {
        if first {
            result.push(c.to_ascii_uppercase());
            first = false;
        } else if c == '_' || c == '-' {
            first = true;
        } else {
            result.push(c);
        }
    }
    result
}

#[test]
fn test_to_camel_case() {
    assert_eq!("helloWorld", to_camel_case("hello-world"));
    assert_eq!("helloWorld", to_camel_case("hello_world"));
    assert_eq!("m2mCoreV1", to_camel_case("m2m-core_v1"));
}

File: src/utils/toCamelCase.ts
export const toCamelCase = (str: string) => {
  let first = false;
  let result = "";

  for (let i = 0; i < str.length; i++) {
    const s = str[i];
    if (i === 0) {
      result += s.toLowerCase();
    } else if (first) {
      result += s.toUpperCase();
      first = false;
    } else if (s === "-" || s === "_") {
      first = true;
    } else {
      result += s;
    }
  }

  return result;
};
